From 338125b0c8765fa50f97a9fb1a10c11cc8ee50d4 Mon Sep 17 00:00:00 2001
From: xxc3nsoredxx <xxc3ncoredxx@gmail.com>
Date: Sat, 9 Jan 2021 21:17:42 -0600
Subject: [PATCH] /etc/issue no initial newline

---
 libbb/login.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/libbb/login.c b/libbb/login.c
index 7f593d80d..d3010162d 100644
--- a/libbb/login.c
+++ b/libbb/login.c
@@ -28,8 +28,6 @@ void FAST_FUNC print_login_issue(const char *issue_file, const char *tty)
 	time(&t);
 	uname(&uts);
 
-	puts("\r");  /* start a new line */
-
 	fp = fopen_for_read(issue_file);
 	if (!fp)
 		return;
-- 
2.26.2

